libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -874,22 +970,22 @@ uint16_t x, usDataLength, usQuestions;
 	{
 		return dnsPARSE_ERROR;
 	}
-	else
-	{
-		xSourceBytesRemaining = xBufferLength;
-	}
+
+	xSourceBytesRemaining = xBufferLength;
 
 	/* Parse the DNS message header. */
 	pxDNSMessageHeader = ( DNSMessage_t * ) pucUDPPayloadBuffer;
 
-	if( pxDNSMessageHeader->usIdentifier == ( uint16_t ) xIdentifier )
+	/* Introduce a do {} while (0) to allow the use of breaks. */
```

Comment:
If you really need to jump out of deeply nested code, say for an error condition, I think this is the one remaining legitimate use for goto.  I believe MISRA allows this use.  An explicit label name can do a lot to clarify an intention.

Reply:
Thanks for this comment. I will keep that in mind when working with pc-lint and the MISRA rules.
But... in some of the +TCP sources, there are long `do {} while(0)` loops with several breaks, thus requiring many goto's.